added skeleton for Login and Sign Up screens 2

dennistech opened the pull request:
- created Signup/Login, Signup, and Login scenes
- set up react-native-router-flux to navigate between scenes
- styled buttons and inputs for each scene
- action/reducers set up for inputs for authenticaion
- added react-native-linear-gradient library for gradient background

Commit:
added skeleton for Login and Sign Up screens 2

philipmok commented on .eslintrc:
```diff
@@ -1,3 +1,16 @@
 {
-  "extends": "airbnb"
+  "extends": [
+    "airbnb"
```
simply `"extends": "airbnb"` like you had before is fine since you are only extending one config

dennistech replied:
ok

philipmok commented on src/Router.js:
```diff
@@ -0,0 +1,39 @@
+          onRight={() => {}}
+          rightTitle="SKIP"
+          key="signUpLogin"
+          component={SignUpLogin}
```
letter casing for `key` and `component` are the same for all your scenes except this one

dennistech replied:
actually messed up the other 2.
ok

philipmok commented on src/Router.js:
```diff
@@ -0,0 +1,39 @@
+          navigationBarStyle={{ backgroundColor: '#007fff' }}
+          titleStyle={{ color: 'white' }}
+        />
+        <Scene
```
since all your scenes share a lot of the same properties, maybe its a good idea to use `map()`, to generate these `<Scene />` components like you would any other list in react.

dennistech replied:
can u find me an example of this, not sure i know what you mean.

philipmok commented on src/components/auth/LogIn.js:
```diff
@@ -0,0 +1,94 @@
+            label="EMAIL"
+            placeholder="HELLO@WOLRD.COM"
+            value={this.props.email}
+            onChangeText={this.onEmailChange}
```
its a better idea to store `onChange` value to component state instead of redux(app state).
u should get `value` from `this.props.email` and set it to component state if a value exist (e.g. fetching this users data from an api).

probably unlikely for this particular scenario, but for any other forms in the project ... if you grab the value for `this.props.auth.email` in a different component, that value may have changed without you knowing due to the user modifying this field.

philipmok commented on src/components/auth/SignUp.js:
```diff
@@ -0,0 +1,125 @@
+          <Input
+            label="EMAIL"
+            placeholder="HELLO@WOLRD.COM"
+            value={this.props.email}
```
i guess what i said above does apply here. if you modify the value in the login component, that will show up here in the SignUp component, which is unusual behavior.

philipmok commented on src/components/auth/SignUpLogin.js:
```diff
@@ -0,0 +1,75 @@
+            <Button
+              backgroundColor="#007fff"
+              fontColor="black"
+              onPress={this.buttonSignUpPressed}
```
for an onPress event with a simple single logic, my preference is to keep it inline

dennistech replied:
ok

philipmok reviewed (commented):
here's some initial feedback for you

philipmok commented on src/components/auth/Input.js:
```diff
@@ -0,0 +1,57 @@
+        keyboardType={keyboardType}
+        secureTextEntry={secureTextEntry}
+        placeholder={placeholder}
+        placeholderTextColor="white"
```
can you make this color `rgba(255,255,255,0.5)`

dennistech replied:
ok

Commit:
- fixed letter casing for 'key' properties in Router.js
- map() through array of properties to generate <Scene /> components
- changed from storing input values in app state to component state for Login.js and SignUp.js
- changed to having onPress inline in SignUpLogin.js
- changed input text placeholder to 50% opacity

- fixed styling ToS background styling
- added white color to Scene back buttons

philipmok commented on src/Router.js:
```diff
@@ -1,36 +1,53 @@
-          titleStyle={{ color: 'white' }}
-        />
+        {scenesConfig.map(scene => (
+          <Scene {...scene} />
```
hmm... isn't `...commonSceneStyles` better off inside this `<Scene />`?
one time placement here vs adding it twice above (for now) inside `scenesConfig`

dennistech replied:
ok, that does work. it would also apply to signUpLogin too, but i guess it is ok because i hid the nav

dennistech replied:
dang it also, have a type in the signUpLogin key

dennistech replied:
not going to implement suggestion, as signUpLogin doesn't use the common styles

Commit:
typo fix

philipmok reviewed (approved):
looks good. 💯
